pennylane/ops/functions/equal.py

```diff
@@ -356,6 +356,42 @@ def _equal_operators(
     return True
 
 
+@_equal_dispatch.register
+def _equal_paulisentence(
+    op1: qml.pauli.PauliSentence,
+    op2: qml.pauli.PauliSentence,
+    check_interface=True,
+    check_trainability=True,
+    rtol=1e-5,
+    atol=1e-9,
+):
+    if set(op1) != set(op2):
+        return f"The Pauli words in the two PauliSentences differ:\n{set(op1)}\n{set(op2)}"
```

Comment:
Might be more 
```suggestion
        if difference1 := set(op1).difference(set(op2)):
            return f"op1 has {difference1} not present in op2"
        difference2 := set(op2).difference(set(op1))
        return f"op2 has {difference2} not present in op1"
```
A little more work, but might be a little more clear as to the source of the difference when the objects get larger.

Reply:
Added a version that always prints all non-vanishing differences. Thanks!